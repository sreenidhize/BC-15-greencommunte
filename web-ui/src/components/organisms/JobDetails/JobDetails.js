import React from "react";
import PropTypes from "prop-types";
import Grid from "@material-ui/core/Grid";
import { Divider, makeStyles, Paper } from "@material-ui/core";
import JobDetailsHeader from "../../molecules/JobDetailsHeader/JobDetailsHeader";
import Content from "../../molecules/Content/Content";
import Button from "../../atoms/button/Button";

const useStyles = makeStyles((theme) => ({
  root: {
    width: "360px",
    paddingLeft: "30px",
    marginRight: "30px",
    marginTop: "30px",
  },
  buttonsStyle: {
    width: "250px",
    marginTop: "16px",
  },
  columnMargin: {
    marginLeft: "8px",
  },
  divider: {
    marginTop: theme.spacing(7),
    marginBottom: theme.spacing(7),
  },
  greenCommute: {
    position: "absolute",
    bottom: 0,
    alignSelf: "center",
    marginBottom: theme.spacing(4),
  },
  buttonLabel: {
    fontSize: "14px",
    padding: theme.spacing(1, 0),
  },
  Paper: {
    width: 390,
    border: 0,
    borderTop: "solid 1px #e3f3f6",
    height: "calc(100vh - 73px)",
    overflowY: "scroll",
    overflowX: "hidden",
  },
}));

const JobDetails = ({
  url,
  alt,
  jobTitle,
  companyName,
  location,
  contentDescription,
  onClickApply,
  onGreenCommuteClick,
  showFindJobs,
  jobId,
  ...props
}) => {
  const classes = useStyles();
  return (
    <Paper className={classes.Paper} elevation={0} variant="outlined" square>
      <Grid
        container
        data-testid="job-details"
        direction="column"
        className={classes.root}
      >
        <Grid item>
          <JobDetailsHeader
            url={url}
            alt={alt}
            jobTitle={jobTitle}
            jobId={jobId}
            companyName={companyName}
            location={location}
            onClickApply={onClickApply}
            isPrimary
            showFindJobs={showFindJobs}
            contentDescription={contentDescription}
            itemsRemoved={(id) => props.itemsRemoved(id)}
          />
        </Grid>
        <Grid item container direction="column">
          {contentDescription.map((item, index) => (
            <React.Fragment key={index}>
              <Divider className={classes.divider} variant="middle" />
              <Content description={item.description} heading={item.heading} />
            </React.Fragment>
          ))}
        </Grid>
        <div className={classes.greenCommute} onClick={onGreenCommuteClick}>
          <Button
            variant="contained"
            color="primary"
            type="long"
            label="Green Commute Routes"
          />
        </div>
      </Grid>
    </Paper>
  );
};

JobDetails.propTypes = {
  url: PropTypes.string,
  alt: PropTypes.string,
  jobTitle: PropTypes.string,
  contentDescription: PropTypes.array,
  companyName: PropTypes.string,
  location: PropTypes.string,
};

export default JobDetails;
