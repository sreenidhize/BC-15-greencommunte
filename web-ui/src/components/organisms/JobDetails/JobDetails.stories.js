import React from 'react';
import JobDetails from './JobDetails';
import theme from '../../../utils/theme/theme';
import { ThemeProvider } from '@material-ui/core';

export default {
    title: 'Organisms/Details',
    component: JobDetails,
};


const Template = (args) => <ThemeProvider theme={theme}><JobDetails {...args} /></ThemeProvider>;

export const Details = Template.bind({});
Details.args = {
    url: "https://m.media-amazon.com/images/I/61b2FkbZsEL._AC_SS350_.jpg",
    alt: "BMW logo",
    jobTitle: "User Experience Designer",
    companyName: "BMW",
    location:"Hitech City, Telangana",
    contentDescription: [
        {
            heading: 'Description',
            description: ["Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide."]
        },
        {
            heading: 'What it takes',
            description: [
                "High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision",
                "Experience Design including the core Adobe Creative Suite products",
                "Excellent written and oral communication and presentation skills"
            ]
        }
    ]
};
