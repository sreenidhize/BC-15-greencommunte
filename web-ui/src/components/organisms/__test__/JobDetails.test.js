import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import JobDetails from "../JobDetails/JobDetails";

describe("Job details", () => {
    it("Matches the snapshot", () => {
        const { container } = render(
            <JobDetails contentDescription={[]}/>
        );
        expect(container).toMatchSnapshot();
    });

    test("test Job details render", () => {
        const { getByTestId } = render(<JobDetails  contentDescription={[]}/>);
        expect(getByTestId("job-details")).toBeInTheDocument();
    });
});