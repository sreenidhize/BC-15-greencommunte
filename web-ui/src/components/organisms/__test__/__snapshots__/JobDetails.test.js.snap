// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job details Matches the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-Paper-7 MuiPaper-outlined"
  >
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column"
      data-testid="job-details"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiGrid-root makeStyles-root-10"
          data-testid="jobDetailsHeader-placeholder"
        >
          <div
            data-testid="img-placeholder"
          >
            <img
              class="makeStyles-secondary-12"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-columnMargin-9 MuiGrid-container MuiGrid-direction-xs-column"
          >
            <h4
              class="MuiTypography-root makeStyles-primary-13 MuiTypography-h4"
              data-testid="cardLabel-placeholder"
            />
            <h4
              class="MuiTypography-root makeStyles-secondary-14 MuiTypography-h4"
              data-testid="cardLabel-placeholder"
            />
            <h4
              class="MuiTypography-root makeStyles-secondary-14 MuiTypography-h4"
              data-testid="cardLabel-placeholder"
            />
            <div
              class="MuiGrid-root makeStyles-buttonsStyle-8 MuiGrid-container MuiGrid-justify-xs-space-between"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-small-16 makeStyles-common-19 MuiButton-outlinedPrimary"
                data-testid="button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Remove
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-small-16 makeStyles-common-19 MuiButton-containedPrimary"
                data-testid="button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Apply
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
      />
      <div
        class="makeStyles-greenCommute-5"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-long-18 makeStyles-common-19 MuiButton-containedPrimary"
          data-testid="button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Green Commute Routes
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
